release-check.sh

```diff
@@ -0,0 +1,22 @@
+#!/usr/bin/env bash
+
+die () { echo "${1}"; exit 1; }
+
+# Check that the current HEAD is on origin/master.
+HEAD=`git rev-parse --verify HEAD`
+MASTERURL="https://api.github.com/repos/facebook/reason/git/refs/heads/master"
+HEADMSG="Current HEAD is not on upstream master. This is a requirement before releasing.
+If you are sure it is, try switching branches to master and pulling changes."
+echo "Checking HEAD against upstream master..."
+curl --silent $MASTERURL | grep "sha" | grep $HEAD || die "$HEADMSG"
```

Comment:
It looks like our current release workflow makes a version on top of master and pushes that. If we want to support arbitrary commits for a release, then we would have to change the Makefile as well. Perhaps a future project.